Code/Mantid/Framework/PythonInterface/plugins/algorithms/DNSDetEffCorrVana.py

```diff
@@ -0,0 +1,244 @@
+            vmean = api.mtd[self.vana_mean_name]
+            if vmean.getNumDims() != nd:
+                message = "Specified VanadiumMean Workspace has wrong dimensions! Must be 2."
+                self.log().error(message)
+                return None
+            if vmean.getNumberHistograms() != 1:
+                message = "Specified VanadiumMean Workspace has wrong number of histograms! Must be 1."
+                self.log().error(message)
+                return None
+            if vmean.blocksize() != 1:
+                message = "Specified VanadiumMean Workspace has wrong number of bins! Must be 1."
+                self.log().error(message)
+                return None
+
+        else:
+            nh = self.vanaws.getNumberHistograms()
+            vmean = api.SumSpectra(ws, IncludeMonitors=False) / nh
+
+        return vmean
+
+    def _cleanup(self, wslist):
+        """
+        deletes workspaces from list
+        """
+        for w in wslist:
+            api.DeleteWorkspace(w)
+        return
+
+    def _vana_correct(self):
+        """
+        creates the corrected workspace
+        """
+        wslist = []
+        # 1. normalize Vanadium and Background
+        vana_normws = api.mtd[self.vanaws.getName() + '_NORM']
+        bkg_normws = api.mtd[self.bkgws.getName() + '_NORM']
+        _vana_norm_ = api.Divide(self.vanaws, vana_normws)
+        wslist.append(_vana_norm_.getName())
+        _bkg_norm_ = api.Divide(self.bkgws, bkg_normws)
+        wslist.append(_bkg_norm_.getName())
+        # 2. substract background from Vanadium
+        _vana_bg_ = _vana_norm_ - _bkg_norm_
+        wslist.append(_vana_bg_.getName())
+        # check negative values, throw exception
+        arr = np.array(_vana_bg_.extractY()).flatten()
+        neg_values = np.where(arr < 0)[0]
+        if len(neg_values):
+            self._cleanup(wslist)
+            message = "Background " + self.bkgws.getName() + " is higher than Vanadium " + \
+                self.vanaws.getName() + " signal!"
+            self.log().error(message)
+            raise RuntimeError(message)
+
+        # 3. calculate correction coefficients
+        _vana_mean_ws_ = self._vana_mean(_vana_bg_)
+        if not _vana_mean_ws_:
+            self._cleanup(wslist)
+            return None
+        if not self.vana_mean_name:
+            wslist.append(_vana_mean_ws_)
+        _coef_ws_ = api.Divide(LHSWorkspace=_vana_bg_, RHSWorkspace=_vana_mean_ws_, WarnOnZeroDivide=True)
+        wslist.append(_coef_ws_)
+        # 4. correct raw data (not normalized!)
+        api.Divide(LHSWorkspace=self.dataws, RHSWorkspace=_coef_ws_, WarnOnZeroDivide=True,
+                   OutputWorkspace=self.outws_name)
+        outws = api.mtd[self.outws_name]
+        # cleanup
+        self._cleanup(wslist)
+        return outws
+
+    def _check_properties(self, lhs_run, rhs_run):
+        """
+        checks whether properties match
+        """
```

Comment:
Just in case you're not aware of that.  you could also document parameters and return values using tags like `@param` and `@returns`. I think that in Python algorithms we are not using that as much as in other parts of the code (and as as much as we ideally should).